### PR TITLE
ENYO-5094: Display overflowing children in slide transition

### PR DESCRIPTION
**Pull request by viodragon2**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Transition applies `overflow: hidden` at the same level as `className` prop in `slide` type transition. In cases where the user wants to set `position: absolute`, children won't be visible.
We see this problem in `moonstone/Notification` where it does not display when `open` is set to true. 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Partial revert from #1481 to apply `overflow: hidden` at an outer `div`.

Enact-DCO-1.0-Signed-off-by: Stephen Choi <stephen.choi@lge.com>

### CODE REVIEW

File: packages/ui/Transition/tests/Transition-specs.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ describe('Transition Specs', () => {
 				);
 
 				const expected = key;
-				const actual = wrapped.find('div').at(0).prop('className');
+				const actual = wrapped.find('div').at(1).prop('className');
```
Comment (viodragon2) [inline packages/ui/Transition/tests/Transition-specs.js]: Should we be concerned that it only tests for `type: slide`?

**Comment by viodragon2** [comment]
The y-position of notification container needs to be anchored at 324px from the bottom, so I've used `translateY(50%)` to address the issue. Also, `pointer-events` were removed from `ui/Trasition` and moved to `moonstone/Popup`.

**Comment by ryanjduffy** [comment]
Good fix, @viodragon2!

**Comment by Djspaceg** [comment]
*A detail clarification.* The Notification's center should be anchored 324px from the bottom. Any changes in height will expand outward from that anchor, both up and down, which is why `translateY(50%)` is used. The bottom of the Notification will never be 324px from the bottom of the screen.

**Pull request merged**